core/arch/opnd.h

```diff
@@ -847,12 +921,16 @@ extern const reg_id_t dr_reg_fixer[];
         DR_REG_SPL /**< Start of 8-bit x64-only register enum values*/
 #    define DR_REG_STOP_x64_8 \
         DR_REG_DIL /**< Stop of 8-bit x64-only register enum values */
-#    define DR_REG_START_MMX DR_REG_MM0  /**< Start of mmx register enum values */
-#    define DR_REG_STOP_MMX DR_REG_MM7   /**< End of mmx register enum values */
-#    define DR_REG_START_XMM DR_REG_XMM0 /**< Start of xmm register enum values */
-#    define DR_REG_STOP_XMM DR_REG_XMM15 /**< End of xmm register enum values */
-#    define DR_REG_START_YMM DR_REG_YMM0 /**< Start of ymm register enum values */
-#    define DR_REG_STOP_YMM DR_REG_YMM15 /**< End of ymm register enum values */
+#    define DR_REG_START_MMX DR_REG_MM0   /**< Start of mmx register enum values */
+#    define DR_REG_STOP_MMX DR_REG_MM7    /**< End of mmx register enum values */
+#    define DR_REG_START_XMM DR_REG_XMM0  /**< Start of xmm register enum values */
+#    define DR_REG_STOP_XMM DR_REG_XMM31  /**< End of xmm register enum values */
```

Comment:
Hmm this will have to be split.  Check the uses: have to use 2 loops??

Reply:
No way we can break binary compatibility? DR_REG_START_XMM <= r <= DR_REG_STOP_XMM will become ugly and error prone otherwise. I could rename (non-compatibility change) to DR_REG_START_SSE_XMM, DR_REG_STOp_SSE_XMM, DR_REG_START_AVX512_XMM, DR_REG_STOP_AVX512_XMM

Reply:
I would vote for not breaking compatility in this way.  This would be too short-term of a decision: if more xmm, ymm, or zmm regs are added in a future ISA extension, we'd be right back in the same situation.

For long-term we'd want people using functions like reg_is_xmm() instead of a hardcoded START..STOP range.  Other uses are often to add an index to START: but those will have to check for avx512 support anyway and will probably thus already have a conditional?  So I would say we'd want to move in that direction.  Making _STOP_XMM documented to only cover the pre-AVX512, seems the step least likely to break things.  Existing code is going to have to be augmented to handle the new zmm registers; having to also handle the new "extended xmm" does not seem horrible.

If we were to change sthg it might be to have reg_is_xmm() not include ymm so it could be used here; or just make a new one reg_is_xmm_only() (name??).

Looking at other uses like https://github.com/za1610/project/blob/99824c3a1a5d70ee9a920e9268d44b31de9cbe1f/DynamoRIO/myClient/cppfp.cpp people are using the enum minus _START to get an index into the mcontext.ymm field.  We'd prefer that they used the interface we provide to map enum to mcontext: dr_{get,set}_value.  That code could use dr_get_value_ex() I believe.  (Looks like we actually don't support writing SIMD today: I filed #3504 on that.)